game/game/gamescript.cpp

```diff
@@ -539,7 +539,7 @@ void GameScript::loadVar(Serialize &fin) {
         for (unsigned j = 0; j < size; ++j) {
             fin.read(v);
             if (s != nullptr && !s->is_member() && !s->is_const()) {
-                s->set_int(v, j);
+                s->set_int(v, static_cast<uint16_t>(j));
```

Comment:
Also in general OpenGothic code prefer to use `size_t` to address array elements. I'm aware, that there is a limit in Daedalus-VM on array size, but is it useful to expose such in api?

Reply:
> Also in general OpenGothic code prefer to use `size_t` to address array elements. I'm aware, that there is a limit in Daedalus-VM on array size, but is it useful to expose such in api?

Yes I think it is. Not right now necessarily but since I will be providing APIs for creating scripts, I expect these values to be used manually. In that case, exposing the limits of the internal representation is useful for indicating what kinds/ranges of values are expected.